packages/gatsby/src/utils/webpack.config.js

```diff
@@ -188,8 +184,10 @@ module.exports = async (
       // optimizations for React) and what the link prefix is (__PATH_PREFIX__).
       plugins.define({
         ...processEnv(stage, `development`),
-        __PATH_PREFIX__: JSON.stringify(
-          program.prefixPaths ? store.getState().config.pathPrefix : ``
+        __BASE_PATH__: JSON.stringify(program.prefixPaths ? pathPrefix : ``),
```

Comment:
Worth adding a note to explain why `base_path = pathPrefix` and `path_prefix = publicPath`? Pulling from your [earlier comment](https://github.com/gatsbyjs/gatsby/pull/12128/files#r279947544), something like:

>BASE_PATH is the new PATH_PREFIX. PATH_PREFIX is now overloaded with assetPrefix + pathPrefix. More info at  https://github.com/gatsbyjs/gatsby/pull/12128